src/components/Tabs/Annotation/CreateLabelForm.tsx

```diff
@@ -0,0 +1,93 @@
+import { Color as AntdColor } from "@rc-component/color-picker";
+import { Button, ColorPicker, ConfigProvider, Input, InputRef } from "antd";
+import React, { ReactElement, useEffect, useRef, useState } from "react";
+import { Color, ColorRepresentation } from "three";
+
+import { FlexColumn, FlexRow } from "../../../styles/utils";
+
+import { LabelOptions } from "../../../colorizer/AnnotationData";
+import { Z_INDEX_POPOVER } from "../../AppStyle";
+import { SettingsContainer, SettingsItem } from "../../SettingsContainer";
+import { DEFAULT_LABEL_COLOR_PRESETS } from "./LabelEditControls";
+
+type CreateLabelFormProps = {
+  initialLabelOptions: LabelOptions;
+  onConfirm: (options: LabelOptions) => void;
+  onCancel: () => void;
+  onColorChanged?: (color: Color) => void;
+  confirmText?: string;
+  focusNameInput?: boolean;
+  zIndex?: number;
+};
+
+const defaultProps: Partial<CreateLabelFormProps> = {
+  confirmText: "Create",
+  focusNameInput: true,
+  onColorChanged: () => {},
+  zIndex: Z_INDEX_POPOVER,
+};
+
+export default function CreateLabelForm(inputProps: CreateLabelFormProps): ReactElement {
+  const props = { ...defaultProps, ...inputProps } as Required<CreateLabelFormProps>;
+
+  const [color, setColor] = useState(props.initialLabelOptions.color);
+  const [nameInput, setNameInput] = useState(props.initialLabelOptions.name);
+  const nameInputRef = useRef<InputRef>(null);
+
+  useEffect(() => {
+    if (nameInputRef.current && props.focusNameInput) {
+      nameInputRef.current.focus();
+    }
+  }, [props.focusNameInput]);
+
+  const confirm = (): void => {
+    props.onConfirm({
+      color: color,
+      name: nameInput.trim(),
+    });
+  };
+
+  const onColorPickerChange = (_color: unknown, hex: string): void => {
+    const newColor = new Color(hex as ColorRepresentation);
+    setColor(newColor);
+    props.onColorChanged(newColor);
+  };
+
+  return (
+    <FlexColumn style={{ width: "250px" }} $gap={10}>
+      <SettingsContainer>
+        <SettingsItem label="Name">
+          <Input
+            value={nameInput}
+            onChange={(e) => setNameInput(e.target.value)}
+            onPressEnter={confirm}
```

Comment:
Curious: does this component need to control the input name state, or could it be pulled out of the `onPressEnter` event instead?